zyte_common_items/pages.py

```diff
@@ -44,18 +45,24 @@ class HasMetadata(Generic[MetadataT]):
     @property
     def metadata_cls(self) -> Type[MetadataT]:
         """Metadata class."""
-        return _get_metadata_class(self)
-
-
-def _get_metadata_class(obj):
-    for base in getattr(obj.__class__, "__orig_bases__", []):
-        origin = getattr(base, "__origin__", None)
-        if not origin:
-            continue
-        if origin != HasMetadata:
-            continue
-        return get_args(base)[0]
-    return None
+        return _get_metadata_class(type(self))
+
+
+def _get_metadata_class(cls: type) -> Type[MetadataT]:
```

Comment:
Could you please a few more tests, for this function specifically, cvering the scenarios you considered while writing it?

Reply:
I also wonder if HasMetadata could be made an argument, though no strong opinion